### DISCUSSION TITLE
Extending existing TButton component

### DISCUSSION THREAD

**Discussion by lamisofteu** [pradosoft/prado#813] [Q&A]
Hello, 
I need extend TButton component with "data-toggle" and "data-target" attributes. I created get* and set* functions, modified "addAttributesToRender()" function, but application still can't parse file with .PAGE extension with modified component in format <com:TButton DataToggle="" />.

Can anybody help?

**Comment by drigolin** [comment]
If you extended TButton and created a new component MyButton then in template you need to use <com:MyButton DataToggle=""/> and not TButton

**Comment by lamisofteu** [comment]
Hello, thank you for reply. Because I need this component only for my private purposes, I modified existing component TButton.

**Reply by drigolin** [reply]
I think you simply can get the standard TButton and use Attributes property

<com:TButton Attributes.data-toggle="toggledata" Attributes.data-target="target" />

**Reply by lamisofteu** [reply]
Thank you, I will try it. This may be way.
